src/event/reader.rs

```diff
@@ -389,8 +442,17 @@ impl EventReader {
                 .await
                 .release_segment(&self.id, &segment, &Offset::new(read_offset))
                 .await
-                .expect("Failed to release segment from RG state for reader");
+                .map_err(|e| {
+                    Error::new(
```

Comment:
The meaning of this is not clear. It looks here as though this is a catch all error. But the docs of the caller are explicet that the reason for an error is that the reader was already offline when offline or release was invoked.

Reply:
fixed this and reader offline API as well.